libinteract/libinteract.py

```diff
@@ -866,6 +866,157 @@ def calc_cg_fullmatrix(identifiers, idxs, percmat, perco):
     # return the full matrix (square matrix)
     return fullmatrix
 
+def filter_by_chain(chain1, chain2, sec_res, array, array_T):
+    """Takes in a table with 2 residue columns and filters it to 
+    return only rows where the first column contains chain1 and the
+    second column contains chain2. Returns filtered table.
+    """
+
```

Comment:
I did this because I thought it would be more efficient since the `filter_by_chain` function can be called many times but uses the same transpose every time. So if I transposed inside the function, it would do the same operation multiple times, even though it only needs to be done once. Should I still change it so that it is transposed on the fly?

Reply:
It's good you're considering being frugal, but in this case yes. Transposing a `numpy.array` returns a view of the array, which is a very inexpensive operation that doesn't change the memory layout (i.e. a proper copy of the array isn't created). So in this case we have a really tiny performance/memory footprint penalty for better readability and maintainability, which I think is worth it